src/providers/hana/qgshanaresultset.cpp

```diff
@@ -0,0 +1,142 @@
+  , mMetadata( mResultSet->getMetaData() )
+{
+}
+
+QgsHanaResultSetRef QgsHanaResultSet::create( StatementRef &stmt, const QString &sql )
+{
+  QgsHanaResultSetRef ret( new QgsHanaResultSet( stmt->executeQuery( reinterpret_cast<const char16_t *>( sql.unicode() ) ) ) );
+  return ret;
+}
+
+QgsHanaResultSetRef QgsHanaResultSet::create( PreparedStatementRef &stmt )
+{
+  QgsHanaResultSetRef ret( new QgsHanaResultSet( stmt->executeQuery() ) );
+  return ret;
+}
+
+void QgsHanaResultSet::close()
+{
+  mResultSet->close();
+}
+
+bool QgsHanaResultSet::next()
+{
+  return mResultSet->next();
+}
+
+QVariant QgsHanaResultSet::getValue( unsigned short columnIndex )
+{
+  switch ( mMetadata->getColumnType( columnIndex ) )
+  {
+    case SQLDataTypes::Bit:
+    case SQLDataTypes::Boolean:
+      return QgsHanaUtils::toVariant( mResultSet->getBoolean( columnIndex ), QVariant::Bool );
+    case SQLDataTypes::Char:
+      return QgsHanaUtils::toVariant( mResultSet->getByte( columnIndex ) );
+    case SQLDataTypes::WChar:
+      return QgsHanaUtils::toVariant( mResultSet->getUByte( columnIndex ) );
+    case SQLDataTypes::SmallInt:
+      if ( mMetadata ->isSigned( columnIndex ) )
+        return QgsHanaUtils::toVariant( mResultSet->getShort( columnIndex ) );
+      else
+        return QgsHanaUtils::toVariant( mResultSet->getUShort( columnIndex ) );
+    case SQLDataTypes::Integer:
+      if ( mMetadata ->isSigned( columnIndex ) )
+        return QgsHanaUtils::toVariant( mResultSet->getInt( columnIndex ) );
+      else
+        return QgsHanaUtils::toVariant( mResultSet->getUInt( columnIndex ) );
+    case SQLDataTypes::BigInt:
+      if ( mMetadata ->isSigned( columnIndex ) )
+        return QgsHanaUtils::toVariant( mResultSet->getLong( columnIndex ) );
+      else
+        return QgsHanaUtils::toVariant( mResultSet->getULong( columnIndex ) );
+    case SQLDataTypes::Double:
+    case SQLDataTypes::Decimal:
+    case SQLDataTypes::Numeric:
+      return QgsHanaUtils::toVariant( mResultSet->getDouble( columnIndex ) );
+    case SQLDataTypes::Date:
+    case SQLDataTypes::TypeDate:
+      return QgsHanaUtils::toVariant( mResultSet->getDate( columnIndex ) );
+    case SQLDataTypes::Time:
+    case SQLDataTypes::TypeTime:
+      return QgsHanaUtils::toVariant( mResultSet->getTime( columnIndex ) );
+    case SQLDataTypes::Timestamp:
+    case SQLDataTypes::TypeTimestamp:
+      return QgsHanaUtils::toVariant( mResultSet->getTimestamp( columnIndex ) );
+    case SQLDataTypes::VarChar:
+    case SQLDataTypes::LongVarChar:
+      return QgsHanaUtils::toVariant( mResultSet->getString( columnIndex ) );
+    case SQLDataTypes::WVarChar:
+    case SQLDataTypes::WLongVarChar:
+      return QgsHanaUtils::toVariant( mResultSet->getNString( columnIndex ) );
+    case SQLDataTypes::Binary:
+    case SQLDataTypes::VarBinary:
+    case SQLDataTypes::LongVarBinary:
+      return QgsHanaUtils::toVariant( mResultSet->getBinary( columnIndex ) );
+    default:
```

Comment:
`SQLDataTypes::TinyInt` is missing. That's the proper type for a single byte.

Reply:
Done in 4d36f6f.